__tests__/integration/utils/wallet-helper.js

```diff
@@ -108,6 +176,44 @@ export class WalletHelper {
     return addressAt;
   }
 
+  /**
+   * Retrieves address information based on the address index inside the wallet.
+   * This is very close to the tests on `address-info.test.js` and as such should reflect any
+   * changes that are made to the calls there.
+   * @param {number} index Address index
+   * @param {string} [token] Token hash, defaults to HTR
+   * @returns {Promise<{
+   * token: (string), index: (number),
+   * total_amount_received: (number), total_amount_sent: (number),
+   * total_amount_locked: (number), total_amount_available: (number)
+   * }>}
+   */
+  async getAddressInfo(index, token) {
+    const response = await TestUtils.request
+      .get('/wallet/address-info')
+      .query({
+        address: await this.getAddressAt(index),
+        token
+      })
+      .set({ 'x-wallet-id': this.#walletId });
+
+    // An error happened
+    if (response.status !== 200 || response.body.success !== true) {
+      throw new Error(`Failure on /wallet/address-info: ${response.text}`);
+    }
+
+    // Returning explicitly each property to help with code completion / test writing
+    const addrInfo = response.body;
+    return {
```

Comment:
For the javascript interpreter it's the same.

For the developer writing the tests, if using an IDE, the code completion works better. On each `expect(addr.` , all the relevant properties are readily shown for easy selection.

Should we decide to use _typescript_ sometime in the future, this would be unnecessary, but for these tests it proved to be a nice writing helper.